scRNA-seq-advanced/02-celltype_assignment.Rmd

```diff
@@ -309,284 +310,453 @@ Note that while we applied this technique to assign cell types using the ADT dat
+We will provide it with the matrix we need, and then a vector of the cluster assignments for each column of the matrix.
+The function will then sum expression values for each gene across all of the columns that have that value.
 
-```{r}
-sce_cluster_mat <- DelayedArray::colsum(counts(sce), sce$nn_cluster)
-dim(sce_cluster_mat)
+```{r sum clusters}
+# sum count matrix by cluster
+cluster_mat <- DelayedArray::colsum(counts(sce), sce$nn_cluster)
+# print new dimensions
+dim(cluster_mat)
 ```
 
-```{r}
-singler_result_cluster <- SingleR::classifySingleR(
-  sce_cluster_mat, 
-  singler_finemodel
+You can see that the resulting matrix still has the same number of rows we have seen before, but now only has rows for the number of clusters that were assigned.
+
+Now we can apply the same `SingleR` model to these results, using the new matrix as input along with the previously trained model.
+As there are only 20 clusters to classify, this will be very quick, and we don't need to parallelize it!
+
+```{r singler cluster, live=TRUE}
+# run SingleR classification with previously trained model
+singler_cluster <- SingleR::classifySingleR(
+  cluster_mat, # cluster expression matrix
+  singler_finemodel # pre-trained model
 )
-```
 
-```{r}
-singler_result_cluster$labels
+# view results
+head(singler_cluster)
 ```
-Add cluster labels to individual cells.
 
-```{r}
-sce$celltype_cluster <- singler_result_cluster$labels[sce$nn_cluster]
+The result is a fairly small table of results, but we are most interested in the labels, which we would like to associate with each cell in our SCE object for visualization.
+Since the cluster labels are the row names of that table, we can perform a cute little trick to assign labels back to each cell based on the name of the cluster that it was assigned to. 
+(In this case the cluster names are all numbers, but that might not always be the case.)
+We'll select values repeatedly from the `singler_cluster` table, using the cluster assignment to pick a row, and then always picking the `pruned.labels` column.
+
+```{r assign cell labels}
+sce$celltype_cluster <- singler_cluster[sce$nn_cluster, "pruned.labels"]
 ```
 
-Plot with cluster labels.
+Now we can plot these cluster-based cell type assignments using the now familiar `plotUMAP()` function
 
-```{r}
+```{r plot cluster celltypes, live=TRUE}
 scater::plotUMAP(sce, colour_by = "celltype_cluster")
 ```
 
-This sure looks nice, but what have we done here? 
-We are _assuming_ that each cluster has only a single cell type!
+This sure looks nice, but what have we really done here? 
+We are _assuming_ that each cluster has only a single cell type, which is a pretty bold assumption, as we really aren't sure that the clusters we created were correct.
+You may recall that clustering algorithms are quite sensitive to parameter choice, so a different parameter choice could quite likely give a different result.
 
+### MetaCell approaches
```

Comment:
I think this is good content, but just want to note we _could_ end up hitting fatigue/time limits for this section. That said the chunks are not live (good!), so this may end up serving as a useful reference for them. Will be interesting to see how it all goes!